### PR TITLE
Import BPABI32 into this repository.

### PR DESCRIPTION
**Pull request by statham-arm**
This spec previously contained six low-resolution and blurred PNG
diagrams, all of which I've redrawn as SVG in Inkscape. As with
previous specs, Inkscape's own versions of the SVG files aren't
correctly handled by rst2pdf (it loses the heads of all the arrows),
so each file comes in a conveniently editable version (named like
"something-source.svg") and a simplified version (named just
"something.svg") that is referred to by the RST document. Editing
these diagrams is best done by editing the source version and then
re-running tools/inkscape/export-svg.sh to regenerate the exported
version.

### CODE REVIEW

File: bpabi32/bpabi32.rst [hunk 1]
```diff
@@ -0,0 +1,2955 @@
...
+
+The figure below depicts a typical set of components present in an
+executable file. Not all components are present in all platform-specific
+formats, and the encoding of dynamic linking data varies greatly between
+platforms. (The dynamic components shown in the figure are a mélange of DLL
+and SVr4 components).
+
+.. rubric:: Components of a typical executable file and the relationships between them
+
+.. figure:: bpabi32-exe-file.svg
```
Comment (stuij) [inline bpabi32/bpabi32.rst:1263]: In the svg, the "Dynamic components (contain dynamic linking data)" text is placed halfway out of the brown box, also making it a bit harder to read.

File: bpabi32/bpabi32.rst [hunk 2]
```diff
@@ -0,0 +1,2955 @@
...
+  This can be done by processing the dynamic relocation directives.
+
+  (Aside: In the RWPI procedure call standard variant [AAPCS32_] – used by
+  Palm OS – this requires a static linker option to retain appropriate
+  static relocation directives in the dynamic relocation section. End
+  aside).
+
+* Adjust the ZI description and append any ZI import tables to it.
+
+The Base Platform ABI Specification
```
Comment (stuij) [inline bpabi32/bpabi32.rst:1652]: missing page break

File: bpabi32/bpabi32.rst [hunk 3]
```diff
@@ -0,0 +1,2955 @@
...
+  processes).
+
+* At the end of the RW program segment (if it must be private to a
+  process).
+
+(Aside: These details are platform-specific and depend on whether the DSO
+will be ROM-resident, ROM-resident and patchable, or only loaded into RAM.
+End aside).
+
+Summary of Platform-Specific Considerations
```
Comment (stuij) [inline bpabi32/bpabi32.rst:2570]: missing page break

File: bpabi32/bpabi32.rst [hunk 4]
```diff
@@ -0,0 +1,2955 @@
...
+
+   3. The RO and RW segments of a DLL are independently loadable. An SVr4
+      DSO has a single base address.
+
+   .. _bpabi32-note4:
+
+   4. This is so that a post linker can place generated RW data between the
+      RW region and the ZI region.
+
+Some Post Linking Sketches
```
Comment (stuij) [inline bpabi32/bpabi32.rst:2782]: missing page break

File: bpabi32/bpabi32.rst (hunk 1 above)
Comment (statham-arm) [reply bpabi32/bpabi32.rst:1263]: Good catch. I got into a muddle with Inkscape and had to reset a bunch of font settings at the last minute, and the UI for doing that also resets the text alignments unwantedly and I had to put them all back by hand. Missed one.

File: bpabi32/bpabi32.rst [hunk 5]
```diff
@@ -0,0 +1,2967 @@
...
+   |                                 | w.r.t the base of the RO segment (there is     | segment base [bpabi32-note3_].                                |
+   | RW → RW/ZI                      | only one base address).                        |                                                               |
+   +---------------------------------+------------------------------------------------+---------------------------------------------------------------+
+   | References to ZI                | References to BSS are fully resolved,          | Each reference to ZI must by relocated by a ZI-relative       |
+   |                                 | PC-relative DSO] or absolute, by the static    | dynamic relocation [bpabi32-note4_].                          |
+   |                                 | linker.                                        |                                                               |
+   +---------------------------------+------------------------------------------------+---------------------------------------------------------------+
+
+.. note::
```
Comment (stuij) [inline bpabi32/bpabi32.rst:2768]: In the PDF, the content of the note crosses the outline of the note.

Comment (statham-arm) [reply bpabi32/bpabi32.rst:2768]: I see that, but what can we do about it? I don't know of any RST feature that asks that not to happen. (Surely it wasn't asked for on purpose in the first place.)

Comment (stuij) [reply bpabi32/bpabi32.rst:2768]: the hacky way around this is to put a page-break before the note

**Comment by stuij** [comment]
LGTM. Thanks!

**Pull request merged**